src/ragger/navigator/nano.py

```diff
@@ -0,0 +1,52 @@
+"""
+   Copyright 2022 Ledger SAS
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+from typing import List
+
+from ragger.backend import BackendInterface
+from ragger.firmware import Firmware
+from ragger.navigator import NavigatorInterface, NavigationInstruction
+
+
+class NanoNavigator(NavigatorInterface):
+
+    _ARGS_KEY = 'args'
+
+    def __init__(self, backend: BackendInterface, firmware: Firmware):
+        super().__init__(backend, firmware)
+
+    def navigate(self, instructions: List[NavigationInstruction]) -> None:
+        """
+        Navigate on the device according to a set of navigation instructions provided.
+
+        :param instructions: Set of navigation instructions.
+        :type instructions: List[NavigationInstruction]
+
+        :raises NotImplementedError: If the navigation instruction is not implemented.
+
+        :return: None
+        :rtype: NoneType
+        """
+        for instruction in instructions:
```

Comment:
One way to uncouple the data from the code flow, in order to avoid future very long `if ... elif ... else ...` structure with potential redundant conditions we could miss, would be to create a dictionnary `{ <instruction> : <callback> }` (or whatever structure), then have a code like:
```python
for instruction in instructions:
    if not instruction in CALLBACKS:
        raise NotImplementedError
    CALLBACKS[instruction]()
```

Reply:
(this could be onlined with a raison `lambda` but it would be very unclear)

Reply:
Advantage of having a dedicated `{ <instruction>: <callback> }` structure is that it is very easy to register (add/modify) custom behavior on any instruction.

Reply:
What if the instruction needs parameters? Let's say the `WAIT_1SEC` I've just added calls time.sleep(1). Shoud I create a `def sleep1(): time.sleep(1)`?

Reply:
Or maybe can you think of a way to add parameters? Maybe instructions could be not just an enum, but a class containing an enum and args?

Reply:
Done!